### PR TITLE
[build] Support releasing ForSt

### PR DESCRIPTION
**Pull request by masteryhx**
## What is the purpose of the change

Support releasing ForSt for linux32(glibc and musl-libc), linux64(glibc and musl-libc), aarch64(glibc and musl-libc), ppc64le(glibc and musl-libc), Mac OSX, Windows x64

## Brief change log

- Adding some make targets, release documentation, build/publish tools
- Fix some platform related codes

## Verifying this change

This change is already covered by existing tests and CI.

### CODE REVIEW

File: java/pom.xml.template [hunk 1]
```diff
@@ -19,17 +19,12 @@
...
     <scm>
-        <connection>scm:git:https://github.com/facebook/rocksdb.git</connection>
-        <developerConnection>scm:git:https://github.com/facebook/rocksdb.git</developerConnection>
-        <url>scm:git:https://github.com/facebook/rocksdb.git</url>
+        <connection>scm:git:https://github.com/ververica/ForSt.git</connection>
+        <developerConnection>scm:git:https://github.com/ververica/ForSt.git</developerConnection>
+        <url>scm:git:https://github.com/ververica/ForSt.git</url>
     </scm>
 
     <organization>
```
Comment (fredia) [inline java/pom.xml.template:30]: Should the organization and developers be updated?

File: java/rocksjni/env_flink.cc [hunk 2]
```diff
@@ -20,8 +20,7 @@
 
 #include <jni.h>
 
-#include <vector>
-
+#include "include/org_rocksdb_FlinkEnv.h"
```
Comment (fredia) [inline java/rocksjni/env_flink.cc:23]: It looks like this file doesn't exist? correct me If I'm wrong.

File: Makefile [hunk 3]
```diff
@@ -2325,10 +2327,41 @@ rocksdbjavastaticrelease: rocksdbjavastaticosx rocksdbjava_javadocs_jar rocksdbj
...
 	openssl sha1 java/target/$(ROCKSDB_JAR_ALL) | sed 's/.*= \([0-9a-f]*\)/\1/' > java/target/$(ROCKSDB_JAR_ALL).sha1
 
+forstjavastaticreleasedocker: rocksdbjavastaticreleasedocker
+	# update apache license
+	mkdir -p java/target/META-INF
+	cp LICENSE java/target/META-INF/LICENSE
+	cd java/target;jar -uf $(ROCKSDB_JAR_ALL) META-INF/LICENSE
+
+	# jars to be released
+	$(eval JAR_PREF=rocksdbjni-$(ROCKSDB_MAJOR).$(ROCKSDB_MINOR).$(ROCKSDB_PATCH))
```
Comment (fredia) [inline Makefile]: forstjni?

**Review by fredia — COMMENTED** [review]
@masteryhx Thanks for the PR, I left some minor comments, PTAL.

File: java/rocksjni/env_flink.cc (hunk 2 above)
Comment (masteryhx) [reply java/rocksjni/env_flink.cc:23]: The header file should be generated manually or by make command as you could see other source files in rocksjni.

**Review by fredia — APPROVED** [review]
@masteryhx Thanks for the update, LGTM.

**Pull request merged**